### PR TITLE
Xsiam validate format changes

### PR DESCRIPTION
**Pull request by dantavori**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-4797](https://jira-hq.paloaltonetworks.local/browse/CIAC-4797)
fixes: [CIAC-4802](https://jira-hq.paloaltonetworks.local/browse/CIAC-4802)
fixes: [CIAC-4274](https://jira-hq.paloaltonetworks.local/browse/CIAC-4274)

## Description
- Added schemas to missing XSIAM content types (where it's possible)
- Added support for formatting and validating XSIAM content items (including fixing an issue that caused `validate -a` to skip running existing XSIAM validations)

Content PR for fixing the existing formatting issues: https://github.com/demisto/content/pull/23889/files

## Must have
- [x] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [x] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/common/schemas/trigger.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            sequence:
+              - include: filter_schema
+          AND:
+            type: seq
+            required: false
+            sequence:
+              - include: filter_schema
+
+
+schema;filter_schema:
```
Comment (daryakoval) [inline demisto_sdk/commands/common/schemas/trigger.yml:36]: what ";" mean?

File: demisto_sdk/commands/common/tools.py [hunk 2]
```diff
@@ -1587,6 +1588,12 @@ def find_type_by_path(path: Union[str, Path] = "") -> Optional[FileType]:
         elif PARSING_RULES_DIR in path.parts:
             return FileType.PARSING_RULE
 
+        elif MODELING_RULES_DIR in path.parts:
+            return FileType.MODELING_RULE
+
+        elif CORRELATION_RULES_DIR in path.parts:
+            return FileType.CORRELATION_RULE
```
Comment (daryakoval) [inline demisto_sdk/commands/common/tools.py:1595]: Strange to have it twice, here and here: https://github.com/demisto/demisto-sdk/blob/fbebca74c9c15efa9a4b497b08101e899d40f24d/demisto_sdk/commands/common/tools.py#L1674
Why did you add it? It did not work without it?

**Review by daryakoval — COMMENTED** [review]
Great!
Something strange happens with Modeling Rules. 
Modeling rules consist from 3 files: xif, yml, schema.json
But the file types are: MODELING_RULE, MODELING_RULE_XIF, MODELING_RULE_SCHEMA. 

But I see other content items has yml file type. So, it should be: MODELING_RULE_YML. 
It is not consistent. Maybe we should consider to change it? (Not in this pr, but open a task for it).

File: demisto_sdk/commands/common/tools.py [hunk 3]
```diff
@@ -1587,6 +1588,12 @@ def find_type_by_path(path: Union[str, Path] = "") -> Optional[FileType]:
         elif PARSING_RULES_DIR in path.parts:
             return FileType.PARSING_RULE
 
+        elif MODELING_RULES_DIR in path.parts:
+            return FileType.MODELING_RULE
```
[the remark this answers is not in the document]
Comment (avidan-H) [reply demisto_sdk/commands/common/tools.py:1592]: Dont' think so. I don't think this was being used.

**Review by dantavori — COMMENTED** [review]
regarding your main comment - MODELING_RULE is the yml, schema is the `_schema.json` file. I agree it's a bit confusing - we can change on a separate PR, as you suggested.
